lib/cppcheck.cpp

```diff
@@ -674,6 +813,15 @@ void CppCheck::executeRules(const std::string &tokenlist, const Tokenizer &token
         }
 
         pcre_free(re);
+
+        // Free up the EXTRA PCRE value (may be NULL at this point)
+        if (pcreExtra) {
+#ifdef PCRE_CONFIG_JIT
+            pcre_free_study(pcreExtra);
```

Comment:
This looks somewhat unintuitive to me. Why do you check for PCRE_CONFIG_JIT here, but call pcre_study()unconditionally?

Reply:
Quoted from https://www.pcre.org/original/doc/html/pcrejit.html
```
You have to do two things to make use of the JIT support in the simplest way:
...

  (2) Use pcre_free_study() to free the pcre_extra block when it is
      no longer needed, instead of just freeing it yourself. This ensures that
      any JIT data is also freed.
For a program that may be linked with pre-8.20 versions of PCRE, you can insert
  #ifndef PCRE_STUDY_JIT_COMPILE
  #define PCRE_STUDY_JIT_COMPILE 0
  #endif
so that no option is passed to pcre_study(), and then use something like this to free the study data:
  #ifdef PCRE_CONFIG_JIT
      pcre_free_study(study_ptr);
  #else
      pcre_free(study_ptr);
  #endif
```

Reply:
Agree. pcre_study shall be called with PCRE_STUDY_JIT_COMPILE option set:
```
        pcre_extra * const pcreExtra = pcre_study(re, PCRE_STUDY_JIT_COMPILE, &pcreStudyErrorStr);
```